test/AddressDerivation.js

```diff
@@ -127,10 +141,14 @@ if (isHardhat) {
                     ['uint256', 'bytes32', 'string', 'string', 'uint8', 'bytes'],
                     [MESSAGE_TYPE_DEPLOY_INTERCHAIN_TOKEN, tokenId, tokenName, tokenSymbol, tokenDecimals, minter],
                 );
+                const wrappedPayload = defaultAbiCoder.encode(
+                    ['uint256', 'string', 'bytes'],
+                    [MESSAGE_TYPE_RECEIVE_FROM_HUB, sourceChain, payload],
+                );
```

Comment:
create a helper for encoding ITS hub send and receive message for easier refactoring in the future
```suggestion
                const hubMessage = receiveFromHubMessage(sourceChain, payload);
```

Similarly, helpers from other message types would be convenient too